backend/lib/endpoints/users.js

```diff
@@ -476,6 +484,74 @@ async function routes(app) {
       return new User(userData).save();
     },
   );
+
+  app.get("/unsubscribe", async (req) => {
+    const { token } = req.headers;
+
+    let decoded = {};
+    jwt.verify(token, config.auth.secretKey, (err, payload) => {
+      if (err) {
+        req.log.error(err, `Invalid token for unsubscribing`);
+        throw app.httpErrors.badRequest("token is invalid");
+      }
+      decoded = payload;
+    })
+
+    const { userId, expireDate } = decoded;
```

Comment:
I think the format will be like `{ userId: 'abc', iat: 1603780773, exp: 1606372773 }`. Tried this in the Node console:
```
> const encodedToken = jwt.sign({ userId: 'abc' }, key, { expiresIn: '30d' });
undefined
> const decodedToken = jwt.verify(encodedToken, key);
undefined
> decodedToken
{ userId: 'abc', iat: 1603780773, exp: 1606372773 }
> new Date(decodedToken.exp*1000)
2020-11-26T06:39:33.000Z

Reply:
right, that's from package `fastify-jwt` and when I tried it, I got an error message likes:
```
    const decoded = app.jwt.verify(token, config.auth.secretKey);
    // error message: "verify must be called asynchronous if secret or 
    //                            public key is provided as a callback"
```

Reply:
Oh no that part is fine, what I wanted to highlight was that the `jsonwebtoken` library automatically adds the expiration in the `exp` key. This is how I'm signing the JWT in the Notification Lambda: https://github.com/FightPandemics/FightPandemics/blob/service/1426-notifications-lambda/services/notification/src/helpers/template-builder.js#L95

Reply:
not sure if I imported the `jwt` wrong, how is your `jwt` got imported (or defined) @mannykary

Reply:
Got it Manny, I will update my date checking accordingly